test/src/common/harnessWal.c

```diff
@@ -0,0 +1,196 @@
+#include "build.auto.h"
+
+#include "common/type/object.h"
+#include "harnessWal.h"
+#include "postgres/interface/crc32.h"
+#include "string.h"
+
+XLogRecord *
+hrnGpdbCreateXRecord(uint8_t rmid, uint8_t info, uint32_t body_size, void *body)
+{
+    XLogRecord *record = memNew(SizeOfXLogRecord + body_size);
+    *record = (XLogRecord){
+        .xl_tot_len = (uint32_t) (SizeOfXLogRecord + body_size),
+        .xl_xid = 0xADDE,
+        .xl_len = body_size,
+        .xl_info = info,
+        .xl_rmid = (uint8_t) rmid,
+        .xl_prev = 0xAABB
+    };
+
+    size_t align_size = MAXALIGN(sizeof(XLogRecord)) - sizeof(XLogRecord);
+    memset((char *) record + sizeof(XLogRecord), 0, align_size);
+
+    if (body == NULL)
+    {
+        memset(XLogRecGetData(record), 0xAB, body_size);
+    }
+    else
+    {
+        memcpy(XLogRecGetData(record), body, body_size);
+    }
+
+    uint32_t crc = crc32cInit();
+    crc = crc32cComp(crc, (unsigned char *) XLogRecGetData(record), body_size);
+    crc = crc32cComp(crc, (unsigned char *) record, offsetof(XLogRecord, xl_crc));
+    crc = crc32cFinish(crc);
+    record->xl_crc = crc;
+
+    return record;
+}
+
+void
+hrnGpdbWalInsertXRecord(
+    Buffer *const walBuffer,
+    XLogRecord *record,
+    InsertXRecordParam param,
+    InsertRecordFlags flags)
+{
+    if (param.magic == 0)
+    {
+        param.magic = 0xD07E;
+    }
+
+    if (param.walPageSize == 0)
+    {
+        param.walPageSize = 32768;
```

Comment:
can we get rid of the magic number 32768 and replace with a define or smth else?